scalafix-core/shared/src/main/scala/scalafix/internal/util/PrettyType.scala

```diff
@@ -396,9 +391,12 @@ class PrettyType private (table: SymbolTable, shorten: QualifyStrategy) {
       try {
         Some(toTree(info).asInstanceOf[Stat])
       } catch {
-        case e: NoSuchElementException =>
+        case _: NoSuchElementException =>
           None
         case NonFatal(e) =>
+          e.setStackTrace(e.getStackTrace.filter { e =>
+            e.getClassName.startsWith("scalafix")
+          })
```

Comment:
Why is this necessary?

Reply:
I've replaced this with a `fatalErrors: Boolean` option to control if exceptions are caught or not. It's possible to set `ExplicitResultTypes.fatalWarnings = true` to control the option. There are still too many NoSuchElementException, for example

- https://github.com/scalameta/scalameta/issues/1625
- https://github.com/scalameta/scalameta/issues/1491
- https://github.com/scalameta/scalameta/issues/1504